### ISSUE TITLE
update to latex-tools | tabularx problem...

### ISSUE THREAD

**Issue by cooch17** [MiKTeX/miktex#1507]
latex-tools bundles together lots of packages, including tabularx. Applied the most recent update to latex-tools, and even a simple MWE with a single table fails to compile (whereas rolling back to earlier latex-tools solves the problem). 

For example, if I update latex-tools, and try to compile the following

     \documentclass{article}

      \usepackage{tabularx}    

     \begin{document}

     \begin{center}
       \begin{tabular}{| c | c | c |}
         a & b & c \\ 
         d & e & f \\  
         g & h & i    
       \end{tabular}
      \end{center}
 
     \end{document}

throws the following error

     Command Line:   latex.exe --enable-write18 --src --interaction=errorstopmode --synctex=-1 "simple_mwe_tabular.tex"

     This is pdfTeX, Version 3.141592653-2.6-1.40.26 (MiKTeX 24.4) (preloaded format=latex.fmt)
      \write18 enabled.
      Source specials enabled.
     entering extended mode
     (simple_mwe_tabular.tex
     LaTeX2e <2023-11-01> patch level 1
     L3 programming layer <2024-03-14>
     (C:\Users\egc\Desktop\egc\text\MikTeX\tex/latex/base\article.cls
     Document Class: article 2023/05/17 v1.4n Standard LaTeX document class
     (C:\Users\egc\Desktop\egc\text\MikTeX\tex/latex/base\size10.clo))
     (C:\Users\egc\Desktop\egc\text\MikTeX\tex/latex/tools\tabularx.sty

     LaTeX Warning: You have requested release `2024/06/01' of LaTeX,
                  but only release `2023-11-01' is available.

     (C:\Users\egc\Desktop\egc\text\MikTeX\tex/latex/tools\array.sty

     LaTeX Warning: You have requested release `2024/06/01' of LaTeX,
                    but only release `2023-11-01' is available.

     (C:\Users\egc\Desktop\egc\text\MikTeX\tex/latex/l3backend\l3backend-dvips.def)
     (simple_mwe_tabular.aux)
    ! Undefined control sequence.
     \@tabular ->\leavevmode \UseTaggingSocket 
                                          {tbl/hmode/begin}\hbox \bgroup $\c...
     l.8 \begin{tabular}
                        {| c | c | c |}
     ?

Some of the warnings are 'interesting' (and perhaps informative --  2024/06/01 release of LaTeX?), but I'm not sure. All I know is that if I comment out tabularx (which I don't really need for this example), compiles fine. And, if I roll back pre-updating latex-tools, no problem, even with tabularx.

So, a straight issue with latex-tools (and everything it contains), or some weiird interaction of latex-tools with some of the other updates which I've also had problems with (specifically, ltxbase and l3kernel)? 

I can roll back pre-update to a perfectly working MikTeX install, but I'm generally happiest not getting too far behind on updates.

**Comment by edocevoli** [comment]
I cannot reproduce the issue and thus cannot provide a fix:

```
$ pdflatex test.tex
This is pdfTeX, Version 3.141592653-2.6-1.40.26 (MiKTeX 24.4) (preloaded format=pdflatex.fmt)
 restricted \write18 enabled.
entering extended mode
(test.tex
LaTeX2e <2024-06-01> patch level 1
L3 programming layer <2024-05-27>
(C:\Users\cs\AppData\Local\Programs\MiKTeX\tex/latex/base\article.cls
Document Class: article 2024/02/08 v1.4n Standard LaTeX document class
(C:\Users\cs\AppData\Local\Programs\MiKTeX\tex/latex/base\size10.clo))
(C:\Users\cs\AppData\Local\Programs\MiKTeX\tex/latex/tools\tabularx.sty
(C:\Users\cs\AppData\Local\Programs\MiKTeX\tex/latex/tools\array.sty))
(C:\Users\cs\AppData\Local\Programs\MiKTeX\tex/latex/l3backend\l3backend-pdftex
.def)
No file test.aux.

[1{C:/Users/cs/AppData/Local/MiKTeX/fonts/map/pdftex/pdftex.map}] (test.aux) )<
C:/Users/cs/AppData/Local/Programs/MiKTeX/fonts/type1/public/amsfonts/cm/cmr10.
pfb>
Output written on test.pdf (1 page, 13506 bytes).
Transcript written on test.log.
```

**Comment by cooch17** [comment]
I decided to try a brute-force experiment. I spun up a WIndows VM on one of my Linux boxes, installed MikTeX, applied *all* of the updates, and...everything worked. So, interpreting this to mean that 'something weird;' was out of synchon my main MikTeX installations, I took one of those machines, completely uninstalled MikTeX, re-installed it, all updates applied and...everything works. 

Not sure what got out of synch -- but the errors I was seeing were repeatable across all (5) of the Windows-based machines I use routinely. 

I guess all I need to do now is simply re-install everything on each of those machines (which isn't a huge problem - takes about 15 minutes to run through everything).
